Covid19Radar/Covid19Radar.iOS/Services/ExposureDetectionBackgroundService.cs

```diff
@@ -0,0 +1,142 @@
+﻿/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using BackgroundTasks;
+using Chino;
+using Covid19Radar.Repository;
+using Covid19Radar.Services;
+using Covid19Radar.Services.Logs;
+using Foundation;
+using Xamarin.Essentials;
+
+namespace Covid19Radar.iOS.Services
+{
+    public class ExposureDetectionBackgroundService : AbsExposureDetectionBackgroundService
+    {
+        private static string BGTASK_IDENTIFIER => AppInfo.PackageName + ".exposure-detection";
+
+        private const int TIMEOUT_IN_MILLIS = 60 * 60 * 1000;
+
+        private readonly AbsExposureNotificationApiService _exposureNotificationApiService;
+        private readonly ILoggerService _loggerService;
+
+
+        public ExposureDetectionBackgroundService(
+            IDiagnosisKeyRepository diagnosisKeyRepository,
+            AbsExposureNotificationApiService exposureNotificationApiService,
+            IExposureConfigurationRepository exposureConfigurationRepository,
+            ILoggerService loggerService,
+            IUserDataRepository userDataRepository,
+            IServerConfigurationRepository serverConfigurationRepository
+            ) : base(
+                diagnosisKeyRepository,
+                exposureNotificationApiService,
+                exposureConfigurationRepository,
+                loggerService,
+                userDataRepository,
+                serverConfigurationRepository
+                )
+        {
+            _exposureNotificationApiService = exposureNotificationApiService;
+            _loggerService = loggerService;
+        }
+
+        public override void Schedule()
+        {
+            _loggerService.StartMethod();
+
+            _loggerService.Debug($"BGTASK_IDENTIFIER: {BGTASK_IDENTIFIER}");
+
+            var result = BGTaskScheduler.Shared.Register(BGTASK_IDENTIFIER, null, task =>
+            {
+                _loggerService.Info("Background task has been started.");
+
+                ScheduleBgTask();
+
+                var cancellationTokenSource = new CancellationTokenSource();
+                task.ExpirationHandler = cancellationTokenSource.Cancel;
+
+                _ = Task.Run(async () =>
+                {
+                    try
+                    {
+                        IList<ExposureNotificationStatus> statuses = _exposureNotificationApiService.GetStatusesAsync()
+                            .GetAwaiter().GetResult();
+
+                        bool isUnauthorized = statuses
+                            .Where(status => status.Code == ExposureNotificationStatus.Code_iOS.Unauthorized)
+                            .Count() != 0;
```

Comment:
このパターンは `.Any()` を使うべきです。

```suggestion
                            .Any(status => status.Code == ExposureNotificationStatus.Code_iOS.Unauthorized);
```

Reply:
6a3cbf0 で対応しました！